ocaml/sdk-gen/csharp/autogen/src/HTTP.cs

```diff
@@ -687,43 +687,57 @@ private static Stream DoHttp(Uri uri, IWebProxy proxy, bool nodelay, int timeout
 
                 stream.Flush();
             }
-            while (ReadHttpHeaders(ref stream, proxy, nodelay, timeout_ms));
+            while (ReadHttpHeaders(ref stream, proxy, noDelay, timeoutMs));
 
             return stream;
         }
 
         /// <summary>
         /// Adds HTTP CONNECT headers returning the stream ready for use
         /// </summary>
-        public static Stream HttpConnectStream(Uri uri, IWebProxy proxy, String session, int timeoutMs)
+        public static Stream HttpConnectStream(Uri uri, IWebProxy proxy, string session, int timeoutMs, params string[] additionalHeaders)

```

Comment:
shouldn't we use a dictionary of strings? `CONNECT <uri> HTTP/1.0` is not really a normal header. As a consumer I'd expect to pass headers as name value pairs, not just the string.

Not a biggie so I'm approving anyway. If you agree feel free to change

Reply:
Hmm, now that you mentioned it, how about using an optional System.Net.WebHeaderCollection object? This might be more flexible as it allows adding headers either as a string or as a key-value pair.

Reply:
sounds perfect, let's do that 
<img src='https://emojis.slackmojis.com/emojis/images/1643515474/14957/meow_checkmark.png?1643515474' height='30' />

Reply:
It turns out that the WebHeaderCollection does not accept a single string as a header unless it contains a colon - for example, a string like `CONNECT <uri> HTTP/1.0` makes it crash - which means the flexibility is only syntactic and it does enforce entering headers as key-value pairs.